Akatiev register and login

pekpytxd opened the pull request:
Created new tests for the register page and login page

pekpytxd committed:
Merged all register tests in one class

pekpytxd committed:
Added new Register Tests and created UserLoginTest.

pekpytxd committed:
Created new Login Tests, created test for ForgottenPasswordPage

adomanich commented on src/main/java/com/opencart/datamodel/WrongLoginModel.java:
```diff
@@ -0,0 +1,14 @@
+package com.opencart.datamodel;
+
+public class WrongLoginModel {
+    private String wrongEmail = "uiu1@k.o";
+    private String wrongPassword = "12345678";
+
+    public String getWrongEmail() {
+        return wrongEmail;
+    }
+
+    public String getWrongPassword() {
+        return wrongPassword;
+    }
+}
```
I think it is better to create a method, which will return a LoginModel with wrong parameters into LoginRepository.

adomanich commented on src/main/java/com/opencart/steps/ForgottenPasswordPageBL.java:
```diff
@@ -0,0 +1,35 @@
+    public LoginPageBL getLoginPageBL() {
+        return new LoginPageBL();
+    }
+
```
Please, remove extra space

adomanich commented on src/main/java/com/opencart/steps/LoginPageBL.java:
```diff
@@ -69,4 +107,40 @@ private void clickOnLoginButton() {
+    }
+
+    public void verifyLoginWithoutEmail() {
+        String expectedMessage = "Warning: No match for E-Mail Address and/or Password.";
+        String expectedMessage1 = "Warning: Your account has exceeded allowed number of login attempts. Please try again in 1 hour.";
```
Could you, please union expectedMessage and expectedMessage1  into one variable?

adomanich commented on src/main/java/com/opencart/steps/LoginPageBL.java:
```diff
@@ -69,4 +107,40 @@ private void clickOnLoginButton() {
+    }
+
+    public void verifyLoginWithEmptyFields() {
+        String expectedMessage = "Warning: No match for E-Mail Address and/or Password.";
+        String expectedMessage1 = "Warning: Your account has exceeded allowed number of login attempts. Please try again in 1 hour.";
```
Could you, please union expectedMessage and expectedMessage1 into one variable?

adomanich commented on src/main/java/com/opencart/steps/LoginPageBL.java:
```diff
@@ -69,4 +107,40 @@ private void clickOnLoginButton() {
+    public void verifyLoginWithEmptyFields() {
+        String expectedMessage = "Warning: No match for E-Mail Address and/or Password.";
+        String expectedMessage1 = "Warning: Your account has exceeded allowed number of login attempts. Please try again in 1 hour.";
+        Assert.assertTrue((expectedMessage + expectedMessage1).contains(loginPage.getWarning().getText()));
```
Please, add error message

adomanich commented on src/main/java/com/opencart/steps/RegisterPageBL.java:
```diff
@@ -72,23 +72,51 @@ public RegisterPageBL registerNewPersonWithoutEmail() {
+        return this;
+    }
+
+    public RegisterPageBL registerNewPersonWithoutPassword() {
+        RegisterModel registerModel = RegisterModelRepository.getRegisterModel();
+        inputFirstName(registerModel.getFirstName());
+        inputLastName(registerModel.getLastName());
+        inputEmail(registerModel.getEmail());
+        inputTelephone(registerModel.getTelephone());
+        clickPolicyCheckbox();
+        clickOnContinueButton();
+        return this;
+    }
+
+    public RegisterPageBL registerNewPersonWithoutVerifyPassword() {
+        RegisterModel registerModel = RegisterModelRepository.getRegisterModel();
+        inputFirstName(registerModel.getFirstName());
+        inputLastName(registerModel.getLastName());
+        inputEmail(registerModel.getEmail());
+        inputTelephone(registerModel.getTelephone());
```
I believe you can set RegisterModel as a parameter into registerNewPerson() method. So, as a result, you won't need to create a few registerNewPerson() methods for this, you will just put corresponding models from test

pekpytxd committed:
Added WrongLoginModel.java in LoginModel
Removed extra spaces in ForgottenPasswordPageBL.java
united expected messages in LoginPageBL
added error message in LoginPageBL.verifyLoginWithEmptyFields
remade method registerUser in RegisterPageBL
added one more test in UserRegisterTest.java

pekpytxd committed:
Added Login Tests

pekpytxd committed:
Added more tests for Register Page.

adomanich commented on src/main/java/com/opencart/steps/RegisterPageBL.java:
```diff
@@ -16,105 +16,36 @@ public RegisterPageBL() {
         clickOnContinueButton();
+
         return this;
     }
```
Please, make clickPolicyCheckbox() public, and click on it from tests when it needs, instead of creating this method. So, in the test where you call this method, you just won't click on PolicyCheckbox

adomanich commented on src/test/java/com/opencart/tests/userLogin/UserLoginTest.java:
```diff
@@ -0,0 +1,67 @@
+        LoginPageBL loginPageBL = mainPageBL.getHeaderPageBL()
+                .clickOnMyAccountButton()
+                .clickOnLoginButton()
+                .loginWithoutPassword();
```
Please, perform the same improvements for Login flow (Move corresponding models to login repository and use one .login() method putting the corresponding LoginModel as a parameter for it)

pekpytxd committed:
Optimized UserLoginTest

adomanich commented on src/main/java/com/opencart/datamodel/LoginModel.java:
```diff
@@ -1,31 +1,63 @@
-    private final String password = "dfghszdfgh";
-    private final String newPassword = "newPassword";
-    private final String newFirstName = "NewFirstName";
+    private String email = "dfghszdfgh@gmail.com";
+    private String newEmail = "newEmail@gmail.com";
+    private String password = "dfghszdfgh";
+    private String newPassword = "newPassword";
+    private String newFirstName = "NewFirstName";
+    private String wrongEmail = "uiu1@k.o";
+    private String wrongPassword = "12345678";
```
You do not need to define those fields here. Set corresponding values directly into methods within LoginModelRepository class.
Leave only getEmail(), getPassword() getters in this class.

pekpytxd replied:
Done

adomanich commented on src/main/java/com/opencart/repository/LoginModelRepository.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    public static LoginModel getValidLoginModel() {
+        LoginModel loginModel = new LoginModel();
```
You do not need to create a new object here.
Just do the following:
.email("someEmail")
.password("somePassword")

pekpytxd replied:
Done

adomanich commented on src/main/java/com/opencart/steps/LoginPageBL.java:
```diff
@@ -42,6 +44,21 @@ public LoginPageBL loginWithNewEmail() {
+        return this;
+    }
+
+
```
remove extra space

pekpytxd replied:
Done

pekpytxd committed:
Removed extra spaces LoginPageBL
Deleted New objects in LoginModelRepository
Deleted defining of Strings in LoginModel

pekpytxd committed:
Little fix

pekpytxd committed:
Little fixes

adomanich commented on src/main/java/com/opencart/datamodel/LoginModel.java:
```diff
@@ -1,31 +1,38 @@
-    private final String newFirstName = "NewFirstName";
+public class LoginModel {
 
+    private String email = "dfghszdfgh@gmail.com";
+    private String password = "dfghszdfgh";
```
Could you, please, move it to LoginRepositry too?

pekpytxd replied:
Done

pekpytxd committed:
Left only Getters of email and password

sofiialevinska reviewed (approved):
супер. я бачу, що зараз цей пул реквест мерджиться з бренчою девелоп

як будеш мерджитись із мастером обережно пліз - щоб не видалити мій код випадково.
там, де про реєстрацію код - замінюй на свою версію - так. 
а там де щось інше - то залишай версію, що в мастері: 

можливо зроби так. я думаю так найбезпечніше:
1. Запуль зміни з мастера собі. 
git checkout master
git pull
2. створи ще одну бренчу від мастера із своїм ім'ям:
git checkout -b Akatiev_RegisterLoginTests
3. скопіюй весь новий код і заміни старий на тій бренчі (я відкривала паралельно на github код і копіювала в ідеешку) і протестуй, що все працює.
4. запуш на git цю бренчу.
5. створи ще один pull request бренчі в master.

я чекатиму, як ти заллєш в мастер - бо буду користуватись кодом по реєстрації теж для своїх тестів по логуванню.

cheesyy183 reviewed (approved):
Gooooooooooood!
